test/org/sagebionetworks/bridge/services/UploadServiceMockTest.java

```diff
@@ -0,0 +1,42 @@
+package org.sagebionetworks.bridge.services;
+
+import static org.junit.Assert.assertSame;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import org.junit.Test;
+
+import org.sagebionetworks.bridge.dao.UploadDao;
+import org.sagebionetworks.bridge.dynamodb.DynamoUpload2;
+import org.sagebionetworks.bridge.exceptions.BadRequestException;
+import org.sagebionetworks.bridge.models.upload.Upload;
+
+public class UploadServiceMockTest {
+    @Test(expected = BadRequestException.class)
+    public void getNullUploadId() {
+        new UploadService().getUpload(null);
+    }
+
+    @Test(expected = BadRequestException.class)
+    public void getEmptyUploadId() {
+        new UploadService().getUpload("");
+    }
+
+    @Test
+    public void getUpload() {
+        // This is a simple call through to the DAO. Test the data flow.
+
+        // mock upload dao
+        UploadDao mockDao = mock(UploadDao.class);
+        Upload mockUpload = new DynamoUpload2();
+        when(mockDao.getUpload("test-upload-id")).thenReturn(mockUpload);
```

Comment:
is the upload always complete when this call works, or does the upload represent the upload when uploading as well? Is that something that would throw an exception if it were still incomplete?